### PR TITLE
Fix JobID assigned to driver root task

### PR DESCRIPTION
**Pull request by pcmoritz**
## What do these changes do?

This restores pre https://github.com/ray-project/ray/pull/5057 behavior for the JobID assigned to a driver.

## Linter

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: src/ray/common/id.h
Comment (stephanie-wang) [inline src/ray/common/id.h]: Hmm do you think this should be a `static` method of `TaskID`? That did sort of make sense to me.

**Review by stephanie-wang — APPROVED** [review]
Added a small comment about whether to make the method static, but otherwise looks good!

File: src/ray/common/id.h
Comment (pcmoritz) [reply src/ray/common/id.h]: Thanks, I restored it!

**Comment by pcmoritz** [comment]
@jovany-wang @raulchen @zhijunfu Can you have a look if this looks good to you?

**Review by jovany-wang — COMMENTED** [review]
Thanks @pcmoritz , but I don't think it's necessary to do this change. have replied at https://github.com/ray-project/ray/pull/5057/files/7dfcbf6faf58e4a498143b729442200b0fe85208

File: src/ray/raylet/node_manager.cc
Comment (jovany-wang) [inline src/ray/raylet/node_manager.cc]: Hey, @pcmoritz , after this https://github.com/ray-project/ray/pull/5004 merged, the driver id is a worker id ,not a job id. so if we'd like to compute a driver_task_id from the given driver id, the current master code is fine I think.
I really don't know what went wrong with current code. I'd appreciate it very much if you can point the issue out deltailed.

Comment (pcmoritz) [reply src/ray/raylet/node_manager.cc]: Yeah, the way it is on the current master, we have the following:
https://github.com/ray-project/ray/blob/master/python/ray/tests/test_basic.py#L2456, i.e. the job_id inside the task specification is equal to the worker_id, which is inconsistent to the definition you are suggesting (I think).

Comment (jovany-wang) [reply src/ray/raylet/node_manager.cc]: Ok, I see. This's caused by the same value of a driver_id and its related job_id.

This PR https://github.com/ray-project/ray/pull/5110 shortens the JobID to 4 bytes, and keeps WorkerID 20 bytes. So #5110 can resolve this problem, I think. also,  
 we should distinguish driver_id from job_id in `RegisterClientRequest`. 

If you couldn't wait until #5110  merged, feel free to distinguish driver_id from job_id in your PR instead of #5099.

Comment (pcmoritz) [reply src/ray/raylet/node_manager.cc]: Ok, that will work! I have a workaround here: https://github.com/ray-project/ray/pull/5076/files#diff-50e88810cd198c83bbce3317b1bc9924R865, we can update it once https://github.com/ray-project/ray/pull/5110 is merged :)

**Pull request closed without merging**
